### PR TITLE
Vet and Visits Feature

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this feature I did the following

- Added new tables for my many-to-many relationships
- Created the vets, specializations and visits tables
- Made sure that the  many-to- many tables connected to the right data
- Answered the given queries
- Followed git flow
- Updated my readme file

**Query Screenshots**

Screenshot 2023-10-27 113306

Screenshot 2023-10-27 114249

Screenshot 2023-10-27 114606

Screenshot 2023-10-27 115657

Screenshot 2023-10-27 150242

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -52,4 +52,43 @@ UPDATE animals
...
+    ((SELECT id FROM animals WHERE name = 'Plantmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-12-21'),
+    ((SELECT id FROM animals WHERE name = 'Plantmon'), (SELECT id FROM vets WHERE name = 'William Tatcher'), '2020-08-10'),
+    ((SELECT id FROM animals WHERE name = 'Plantmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2021-04-07'),
+    ((SELECT id FROM animals WHERE name = 'Squirtle'), (SELECT id FROM vets WHERE name = 'Stephanie Mendez'), '2019-09-29'),
+    ((SELECT id FROM animals WHERE name = 'Angemon'), (SELECT id FROM vets WHERE name = 'Jack Harkness'), '2020-10-03'),
+    ((SELECT id FROM animals WHERE name = 'Angemon'), (SELECT id FROM vets WHERE name = 'Jack Harkness'), '2020-11-04'),
+    ((SELECT id FROM animals WHERE name = 'Boarmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-01-24'),
+    ((SELECT id FROM animals WHERE name = 'Boarmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-05-15'),
+    ((SELECT id FROM animals WHERE name = 'Boarmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-01-24'),
```
Comment (Ghiftee) [inline data.sql]: - [ ] Well done! You have successfully added all data as requested in the requirements. However, you have one more data than required. 
You are currently inserting the data below twice, kindly remove one.
```
((SELECT id FROM animals WHERE name = 'Boarmon'), (SELECT id FROM vets WHERE name = 'Maisy Smith'), '2019-01-24')
```

File: queries.sql [hunk 2]
```diff
@@ -124,3 +124,43 @@ FROM owners o
...
+SELECT ve.name, s.name 
+from vets ve LEFT JOIN specializations spe ON spe.vet_id = ve.id 
+LEFT JOIN species s ON s.id = spe.species_id;
+
+SELECT a.name, COUNT(*) AS num_visits
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+GROUP BY a.id
+ORDER BY num_visits DESC
+LIMIT 1;
```
Comment (Ghiftee) [inline queries.sql:152]: - [x] This query currently returns `5` but it should return `4`. Kindly correct your table data above and perform the query again. Do not forget to update your screenshot afterwards 👍🏽

File: queries.sql [hunk 3]
```diff
@@ -124,3 +124,43 @@ FROM owners o
...
+SELECT COUNT(DISTINCT a.id) AS num_animals_seen
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN vets ve ON v.vet_id = ve.id
+WHERE ve.name = 'Stephanie Mendez';
+
+SELECT ve.name, s.name 
+from vets ve LEFT JOIN specializations spe ON spe.vet_id = ve.id 
+LEFT JOIN species s ON s.id = spe.species_id;
+
```
Comment (Ghiftee) [inline queries.sql:146]: - [ ] Good job, your queries have been successful so far! You are currently missing the query in the image below which should be the next query. I noticed it is present in your screenshot but you also need to add it to your query file.

### Missing query
Screenshot 2023-10-27 at 11 46 49

### Your screenshot
Screenshot 2023-10-27 at 11 46 49 (2)

File: queries.sql [hunk 4]
```diff
@@ -124,3 +124,43 @@ FROM owners o
...
+SELECT a.name, COUNT(*) AS num_visits
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+GROUP BY a.id
+ORDER BY num_visits DESC
+LIMIT 1;
+
```
Comment (Ghiftee) [inline queries.sql:153]: - [ ] Please do not forget to add the two queries in the image below to your `queries.sql` file. They are both present in your screenshot but also need to be in your code.

Screenshot 2023-10-27 at 11 53 58

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello @Chimoneg27 👋🏽 

Good job working on the project so far! 
To highlight 👍🏽 
✔️ All data are in appropriate files
✔️ Queries are successful
✔️ Screenshots are added

However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review 👇🏽 _

- [ ] It is fantastic that you have added the screenshots of all the queries as required. However, in completely fulfilling all requirements, you need to take it one step further. Kindly add the screenshots to your PR description.

Screenshot 2023-10-27 at 11 21 18


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: queries.sql (hunk 4 above)
Comment (Chimoneg27) [reply queries.sql:153]: Hello gift. About the images. I prefer to put them in a folder. Putting them in my PR would make it too long and a bit tedious to go through.

**Review by ridwanediallo — APPROVED** [review]
# STATUS: APPROVED 🟢
## Hi @Chimoneg27 , 
Congratulations!GIF

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Well done implementing the requested changes. 🎉 🎉 🎉 ✔

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**